pkg/cloud/aggregation.go

```diff
@@ -8,39 +8,64 @@ import (
 	corev1 "k8s.io/api/core/v1"
 )
 
-var aggregationVarNames = []string{
-	"K6_CLOUD_AGGREGATION_MIN_SAMPLES",
-	"K6_CLOUD_AGGREGATION_PERIOD",
-	"K6_CLOUD_AGGREGATION_WAIT_PERIOD",
-	"K6_CLOUD_METRIC_PUSH_INTERVAL",
-	"K6_CLOUD_MAX_METRIC_SAMPLES_PER_PACKAGE",
-	"K6_CLOUD_MAX_METRIC_PUSH_CONCURRENCY",
+var aggregationVarNames = map[int][]string{
+	1: []string{
+		// cloud output v1: to be removed in the future
+		"K6_CLOUD_AGGREGATION_MIN_SAMPLES",
+		"K6_CLOUD_AGGREGATION_PERIOD",
+		"K6_CLOUD_AGGREGATION_WAIT_PERIOD",
+		"K6_CLOUD_METRIC_PUSH_INTERVAL",
+		"K6_CLOUD_MAX_METRIC_SAMPLES_PER_PACKAGE",
+		"K6_CLOUD_MAX_METRIC_PUSH_CONCURRENCY",
+	},
+	2: []string{
+		// cloud output v2
+		"K6_CLOUD_API_VERSION",
+		"K6_CLOUD_AGGREGATION_PERIOD",
+		"K6_CLOUD_AGGREGATION_WAIT_PERIOD",
+		"K6_CLOUD_METRIC_PUSH_INTERVAL",
+		"K6_CLOUD_MAX_METRIC_PUSH_CONCURRENCY",
+	},
 }
 
 func EncodeAggregationConfig(testRun *cloudapi.CreateTestRunResponse) string {
-	return fmt.Sprintf("%d|%s|%s|%s|%d|%d",
-		testRun.ConfigOverride.AggregationMinSamples.Int64,
+	return fmt.Sprintf("%d|%s|%s|%s|%d",
+		2, // use v2 for all new test runs
 		testRun.ConfigOverride.AggregationPeriod.String(),
 		testRun.ConfigOverride.AggregationWaitPeriod.String(),
 		testRun.ConfigOverride.MetricPushInterval.String(),
-		testRun.ConfigOverride.MaxMetricSamplesPerPackage.Int64,
 		testRun.ConfigOverride.MetricPushConcurrency.Int64)
 }
 
 func DecodeAggregationConfig(encoded string) ([]corev1.EnvVar, error) {
 	values := strings.Split(encoded, "|")
-	if len(values) != len(aggregationVarNames) {
+
+	// in order not to break existing deployments,
+	// let's support decoding of cloud output v1 for some time
```

Comment:
What is the expectation here? Isn't versioning of the k6-operator already a solution for it?

Reply:
K6 resource is separate from k6-operator. And while in most cases, there should be a deletion of resources on update, it's not guaranteed AFAIK.